unit/solvers/smt2_incremental/construct_value_expr_from_smt.cpp

```diff
@@ -68,6 +68,16 @@ TEST_CASE("Value expr construction from smt.", "[core][smt2_incremental]")
     rowt{smt_bool_literal_termt{false}, false_exprt{}},
     rowt{smt_bit_vector_constant_termt{0, 8}, from_integer(0, c_bool_typet(8))},
     rowt{smt_bit_vector_constant_termt{1, 8}, from_integer(1, c_bool_typet(8))},
+    rowt{
+      smt_bit_vector_constant_termt{0, 64},
+      from_integer(0, pointer_typet(unsignedbv_typet(64), 64))},
+    // The reason for the more intricate elaboration of a pointer with a value
+    // of 12 is a limitation in the design of from_integer, which only handles
+    // pointers with value 0 (null pointers).
+    rowt{
+      smt_bit_vector_constant_termt{12, 64},
+      constant_exprt(
+        integer2bvrep(12, 64), pointer_typet(unsignedbv_typet(64), 64))},
```

Comment:
:bulb: It could be a bit tidier if these pointer tests were separated out into a macro like for the BIT_VECTOR_TESTS.
:bulb: These tests could be made more thorough by re-running them for 32 and 64 bit pointer widths.
:bulb: Using a pointer to a type such as `bool_typet` or `void_type()` instead of a pointer to `unsignedbv_typet(64)`, may help improve the readability of the test by making all instances of the number `64` relate to the width of the pointer, rather than the width of the type the pointer points to.